src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1756,11 +1758,22 @@ public IOperationResult Compact(Action<IOperationProgress> onProgress)
             lock (_locker)
             {
                 _isCompactionInProgress = true;
+
+                var progress = new IndexCompactionProgress();
+                var currentlyRunningQueriesCount = CurrentlyRunningQueries.Count;
```

Comment:
There is a race here, a user can check the `_isCompactionInProgress` in `AssertIndexState`, see it is valid, and then start querying, and we check if there are running queries before that. 
You can't rely on this